### PR TITLE
BUG: fixes abort trap 6 error in scipy issue 9785 in unit tests

### PR DESCRIPTION
**Pull request by rlucas7**
Fixes issue #9785

### CODE REVIEW

File: scipy/stats/tests/test_morestats.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 try:
     import matplotlib.pyplot as plt
     have_matplotlib = True
+    matplotlib.rcParams['backend'] = 'Agg'
```
Comment (pv) [inline scipy/stats/tests/test_morestats.py]: The rcParam has to be changed before importing pyplot.

Comment (pvanmulbregt) [reply scipy/stats/tests/test_morestats.py]: Split the import into two statements with the parameter setting sandwiched in the middle.
```
    import matplotlib
    matplotlib.rcParams['backend'] = 'Agg'
    import matplotlib.pyplot as plt
    have_matplotlib = True
```

**Review by pvanmulbregt — CHANGES_REQUESTED** [review]

**Comment by rlucas7** [comment]
@pvanmulbregt Thanks! I made the changes. It makes sense to me what you are stating, the backend is (I'm guessing) attached to `plt` at import time so that is why we need to attach before. The travis CI builds fine. The circle/ci build is something to do with a differential evolution solver that must be a flaky test.

**Review by pvanmulbregt — APPROVED** [review]

**Comment by pvanmulbregt** [comment]
This PR successfully fixed the abort trap 6 in my environment without having to resort to setting `backend` in   ~/.matplotlib/matplotlibrc. 
The build_docs failed with a timeout and the pypy3 with an unrelated differential_evolution issue.
LGTM.

**Comment by rgommers** [comment]
In it goes. Thanks @rlucas7, all.

**Pull request merged**